### DISCUSSION TITLE
Faster Remote and Non-Blittable Types

### DISCUSSION THREAD

**Discussion by cmadh** [microsoft/FASTER#715] [General]
Digging through the Faster Repository and the remote-samples I'm wondering if it's currently possible to use FasterServerTcp/FasterKVClient with Non-Blittable Types. 

SubscribeKVBroker and SubscribeBroker seem to work with memory-regions of objects returned by ref so they seem to only work with Blittable-Types, is that correct? 

IKeySerializer<Key>, IKeyInputSerializer<Key, Input>, IServerSerializer<Key, Value, Input, Output> allow any kind of Type as Generic parameter (it's not even limited to structs) and there's no single comment about blittable- and non-blittable-types which confuses me a little. 

I'm also confused by how FasterKV and FasterServerTcp connect to each other as the FasterKVProviderBase registered in a FasterTCPServer uses a FasterKV-instance internally. 

Is there a way to use Non-Blittable Types with the remote-stack at the moment by doing some form of conversion between Non-Blittable Objects and blittable types like Span<byte>?

I would greatly appreciate some clarification or even a small example with a Non-Blittable Type if possible!

**Comment by badrishc** [comment]
The best approach here would be to use VarLenServer and serialize your types into Span<byte> at the client side.

We do not properly support arbitrary types on client rehydrating into the server as the same type, beyond some untested code in GenericServer.

**Reply by cmadh** [reply]
Thank you for the response, that matches my thoughts, we'll give it a try.
